src/Tools/vmOpsTool/IVMWare.java

```diff
@@ -92,10 +93,11 @@ void createSnapshot(String vmName, String snapshotName, boolean saveVMMemory, bo
 
     /**
      * @param vmName   name of the virtual machine
+     * @param isCustomizationRequired indiactes whether this operation need to wait for customization or not.
      * @param connData vCenter connection information
      * @throws Exception
      */
-    void powerOnVM(String vmName, ConnectionData connData) throws Exception;
+    void powerOnVM(String vmName, boolean isCustomizationRequired, ConnectionData connData) throws Exception;

```

Comment:
Is it possible to determine this state from VM properties instead of having to introduce this variable?